kuma/javascript/src/formatters.js

```diff
@@ -0,0 +1,76 @@
+// @flow
+type DateTimeFormatOptions = {
+    localeMatcher?: 'lookup' | 'best fit',
+    timeZone?: string,
+    hour12?: boolean,
+    formatMatcher?: 'basic' | 'best fit',
+    weekday?: 'narrow' | 'short' | 'long',
+    era?: 'narrow' | 'short' | 'long',
+    year?: 'numeric' | '2-digit',
+    month?: 'numeric' | '2-digit' | 'narrow' | 'short' | 'long',
+    day?: 'numeric' | '2-digit',
+    hour?: 'numeric' | '2-digit',
+    minute?: 'numeric' | '2-digit',
+    second?: 'numeric' | '2-digit',
+    timeZoneName?: 'short' | 'long',
+};
+
+type NumberFormatOptions = {
+    localeMatcher?: 'lookup' | 'best fit',
+    style?: 'decimal' | 'currency' | 'percent',
+    currency?: string,
+    currencyDisplay?: 'symbol' | 'code' | 'name',
+    useGrouping?: boolean,
+    minimumIntegerDigits?: number,
+    minimumFractionDigits?: number,
+    maximumFractionDigits?: number,
+    minimumSignificantDigits?: number,
+    maximumSignificantDigits?: number,
+};
+
+/**
+ * Basic date formatting according to locale.
+ * Defaults to long month, numeric day and year
+ * e.g. for en-US, April 22, 2020
+ *
+ * @param {string} locale - locale identifier
+ * @param {Object | string | number} date - date to be formatted
+ * @param {Object} [options] - formatting options
+ * @returns {string} - formatted date string
+ */
+
+export const formatDate = (
+    locale: string,
+    date: Date | number | string,
+    options?: DateTimeFormatOptions = {
+        year: 'numeric',
+        month: 'long',
+        day: 'numeric',
+    }
+): string => {
+    const dateObj = new Date(date);
```

Comment:
Now I get it! If `date` already was a `Date` object, doing `new Date(aDateObj)` is harmless. It just returns a new Date object instance.

Reply:
> I still don't understand or see what the `SignInLink` has to do with anything and why it does the extrac check of `typeof window`.

I removed the `typeof window` check, but just so you know, I copied it from the original login code. It looked like an important check so I copied it over! https://github.com/mdn/kuma/blob/master/kuma/javascript/src/header/login.jsx#L16

Reply:
😸 It is for sure an important check, we should probably change the other instance as well :) Not in this PR though ;)

Reply:
I think the commenting is getting criss-crossed. 

So it *used* to be this:
```javascript
function Login() {
  const userData = useContext(UserProvider.context);
  if (!userData || typeof window === 'undefined') {
    return null;
  }
  if (userData) {
    return <img src={userData.mugshot}>
  } else {
    return <a href="signin" onClick={openModal}>Sign in</a>
  }
```
Then you extracted the `return <a href="signin" onClick={openModal}>Sign in</a>` into its own new component and inside that one you did the check on `typeof window !== 'undefined'` **again**. 

I think that confused me. 
Also, it confused me why you refactored `<Login/>` but you, in some other comment, explained that you wanna reuse `<SigninLink/>` in more places than just within `<Login/>`.

All good here.